Sources/ToggleManager/ToggleManager.swift

```diff
@@ -102,12 +110,14 @@ extension ToggleManager {
         return !options.contains(.noCaching)
     }
     
-    private func isValueValid(value: Value, defaultValue: Value?) -> Bool {
-        if shouldCheckInvalidValueTypes, let defaultValue, !(value ~= defaultValue) {
+    private func isValueValid(value: Value, defaultValue: Value?, variableName: Variable, providerName: String) -> Bool {
+        if shouldCheckInvalidValueTypes, let defaultValue, value.toggleTypeDescription != defaultValue.toggleTypeDescription {
+            errorLogger?.logError(ProviderValueError.invalidValueType("\(variableName) was found with an invalid type (\(value.toggleTypeDescription)) in Provider: \(providerName)"))
```

Comment:
It would be better to avoid hard-coded English strings in code. All we need is to pass the error type for which we have an enum already. That's what brings the information.

```
logger?.log(.invalidValueType(variable, value, provider))
```

Reply:
this would remove the need for `toggleTypeDescription`